icubam/backoffice/handlers/icus.py

```diff
@@ -31,29 +31,37 @@ def get(self):
 class ICUHandler(base.BaseHandler):
   ROUTE = "/icu"
 
-  @tornado.web.authenticated
-  def get(self):
-    icu = self.db.get_icu(self.get_query_argument('id', None))
-    icu = icu if icu is not None else store.ICU()
-    if icu.is_active is None:
-      icu.is_active = True
-
+  def do_render(self, icu, error=False):
     if self.user.is_admin:
       regions = self.db.get_regions()
     if not self.user.is_admin:
       regions = [e.region for e in self.user.managed_icus]
     regions.sort(key=lambda r: r.name)
-    self.render("icu.html", icu=icu, regions=regions, error='')
+
+    icu = icu if icu is not None else store.ICU()
+    if icu.is_active is None:
+      icu.is_active = True
+    return self.render("icu.html", icu=icu, regions=regions, error=error)
 
   @tornado.web.authenticated
-  def post(self):
+  def get(self):
     icu = self.db.get_icu(self.get_query_argument('id', None))
+    return self.do_render(icu)
+
+  @tornado.web.authenticated
+  def post(self):
     values = self.parse_from_body(store.ICU)
-    values["is_active"] = bool(values["is_active"])
-    # TODO(olivier): try catch error here.
-    if icu is None:
-      self.db.add_icu(self.user.user_id, store.ICU(**values))
-    else:
-      self.db.update_icu(self.user.user_id, icu.icu_id, values)
+    values["is_active"] = values["is_active"] == 'True'
+    id_key = 'icu_id'
+    icu_id = values.pop(id_key, '')
+    try:
+      if not icu_id:
+        self.db.add_icu(self.user.user_id, store.ICU(**values))
```

Comment:
not really needed but just in case we want to keep the user in the same page instead of redirecting in line 67 (if people ask us because they want to keep editing or double check what they had done), this will be much safer if we assign the new icu_id:
icu_id = self.db.add_icu(self.user.user_id, store.ICU(**values))

Reply:
done